src/kboot_gpu.c

```diff
@@ -37,6 +38,18 @@ struct aux_perf_states {
     struct aux_perf_state states[];
 };
 
+int32_t rust_gpu_initdata_size(uint32_t compat_maj, uint32_t compat_min, size_t *data_a,
+                               size_t *data_b, size_t *globals);
+
+int32_t rust_fill_gpu_initdata(size_t perf_state_table_count, size_t perf_state_count,
+                               const struct perf_state *c_perf_states, uint32_t *max_pwr,
+                               float *core_leak, float *sram_leak, float *cs_leak, float *afr_leak,
+                               size_t n_perf_states_cs, const struct aux_perf_state *pstates_cs,
+                               size_t n_perf_states_afr, const struct aux_perf_state *pstates_afr,
+                               uint32_t is_j180, uint32_t base_pstate, uint64_t uat_ttb_base,
+                               uint32_t num_cores, uint32_t compat_maj, uint32_t compat_min,
+                               uint32_t gpu_rev_id, void *data_a, void *data_b, void *globals);
```

Comment:
I'd feel more comfortable if we pass the size `data_a`, `data_b` and `globals` as well.

Reply:
The caller is supposed to allocate exactly as much as `rust_gpu_initdata_size` asked for, so not sure what value that adds?